openapiart/generator.py

```diff
@@ -406,17 +406,38 @@ def _serialize_payload(self, payload):
             payload = json.dumps(payload)
         return payload
 
+    def _raise_exception(self, grpc_error):
+        err = self.error()
+        try:
+            err.deserialize(grpc_error.details())
+        except Exception as _:
+            err.code = grpc_error.code().value[0]
+            err.errors = [grpc_error.details()]
+        raise Exception(err)
+
     def from_exception(self, grpc_error):
-        # type: (grpc.RpcError) -> Error
-        err = None
-        if isinstance(grpc_error, grpc.RpcError):
+        # type: (Exception) -> Union[Error, None]
+
+        if isinstance(grpc_error, Error):
+            return grpc_error
+        elif isinstance(grpc_error, grpc.RpcError):
             err = self.error()
-            try:
-                err.deserialize(grpc_error.details())
-            except Exception as e:
-                err.code = 2 # code for unknown error
-                err.errors = [grpc_error.details()]
-        return err
+            err.code = grpc_error.code().value[0]
+            err.errors = [grpc_error.details()]
+            return err
+        else:
+            if len(grpc_error.args) != 1:
```

Comment:
shouldn't we check if `args` is even a valid property ? (since we're not always expecting an Exception object)

Reply:
done